File: core/src/de/project/ice/ecs/Families.java
package de.project.ice.ecs;

import com.badlogic.ashley.core.Family;
import de.project.ice.ecs.components.HotspotComponent;
import de.project.ice.ecs.components.*;
import org.jetbrains.annotations.NotNull;

@SuppressWarnings("unchecked")
public abstract class Families {
    @NotNull
    public static final Family renderable = Family.all(TransformComponent.class, TextureComponent.class).exclude(InvisibilityComponent.class).get();
    @NotNull
    public static final Family animated = Family.all(TransformComponent.class, TextureComponent.class, AnimationComponent.class, StateComponent.class).exclude(InvisibilityComponent.class).get();
    @NotNull
    public static final Family scripted = Family.all(ScriptComponent.class).exclude(DisabledComponent.class).get();
    @NotNull
    public static final Family playerCharacter = Family.all(TransformComponent.class, TextureComponent.class, AnimationComponent.class, StateComponent.class, BreathComponent.class, MovableComponent.class, ControlComponent.class).exclude(InvisibilityComponent.class).get();
    @NotNull
    public static final Family camera = Family.all(CameraComponent.class).exclude(DisabledComponent.class).get();
    @NotNull
    public static final Family breathing = Family.all(BreathComponent.class, IdleComponent.class, StateComponent.class).exclude(InvisibilityComponent.class, DisabledComponent.class).get();
    @NotNull
    public static final Family hotspot = Family.all(HotspotComponent.class, TransformComponent.class).exclude(DisabledComponent.class).get();
    @NotNull
    public static final Family walkArea = Family.all(WalkAreaComponent.class).exclude(DisabledComponent.class).get();
    @NotNull
    public static final Family controllable = Family.all(ControlComponent.class, MovableComponent.class, TextureComponent.class, TransformComponent.class).exclude(DisabledComponent.class).get();
}

File: core/src/de/project/ice/ecs/Families.kt
package de.project.ice.ecs

import com.badlogic.ashley.core.Family
import de.project.ice.ecs.components.*

@SuppressWarnings("unchecked")
object Families {
    @JvmField val renderable = Family.all(TransformComponent::class.java).exclude(InvisibilityComponent::class.java).get()
    @JvmField val animated = Family.all(TransformComponent::class.java, TextureComponent::class.java, AnimationComponent::class.java).exclude(InvisibilityComponent::class.java).get()
    @JvmField val scripted = Family.all(ScriptComponent::class.java).exclude(DisabledComponent::class.java).get()
    @JvmField val playerCharacter = Family.all(TransformComponent::class.java, TextureComponent::class.java, AnimationComponent::class.java, BreathComponent::class.java, ControlComponent::class.java).exclude(InvisibilityComponent::class.java).get()
    @JvmField val camera = Family.all(CameraComponent::class.java).exclude(DisabledComponent::class.java).get()
    @JvmField val breathing = Family.all(BreathComponent::class.java, IdleComponent::class.java).exclude(InvisibilityComponent::class.java, DisabledComponent::class.java).get()
    @JvmField val hotspot = Family.all(HotspotComponent::class.java, TransformComponent::class.java).exclude(DisabledComponent::class.java).get()
    @JvmField val walkArea = Family.all(WalkAreaComponent::class.java).exclude(DisabledComponent::class.java).get()
    @JvmField val controllable = Family.all(ControlComponent::class.java, TextureComponent::class.java, TransformComponent::class.java).exclude(DisabledComponent::class.java).get()
    @JvmField val distanceScale = Family.all(DistanceScaleComponent::class.java, TransformComponent::class.java).get()
    @JvmField val assets = Family.one(TextureComponent::class.java, AnimationComponent::class.java).get()
}
